File: problems/2900-longest-unequal-adjacent-groups-subsequence-i/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
char** getLongestSubsequence(char** words, int wordsSize, int* groups, int groupsSize, int* returnSize) {
    char** result = (char**)malloc(sizeof(char*) * wordsSize);
    int count = 0;
    int lastGroup = -1;
    
    for (int i = 0; i < wordsSize; i++) {
        if (groups[i] != lastGroup) {
            result[count] = words[i];
            count++;
            lastGroup = groups[i];
        }
    }
    
    *returnSize = count;
    return result;
}

File: problems/2900-longest-unequal-adjacent-groups-subsequence-i/solution.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        res = [words[0]]
        prev_group = groups[0]
        for i in range(1, len(words)):
            if groups[i] != prev_group:
                res.append(words[i])
                prev_group = groups[i]
        return res